src/App.tsx

```diff
@@ -15,71 +19,98 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = { selectedGood: string };
+
+export class App extends Component<{}, State> {
+  state = { selectedGood: 'Jam' };
+
+  isSelected = (item: string) => (
+    item === this.state.selectedGood
+  );
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  clearSelected = () => (
+    this.setState({ selectedGood: '' })
+  );
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  setSelected = (item: string) => (
+    this.setState({
+      selectedGood: this.isSelected(item)
+        ? ''
+        : item,
+    })
+  );
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render = (): ReactNode => {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        <h1
+          className={cn(
+            'title', {
+              'is-flex is-align-items-center': selectedGood,
+            },
+          )}
+        >
+          {selectedGood
+            ? (
+              <>
+                {`${selectedGood} is selected`}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+                <button
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  onClick={this.clearSelected}
+                />
+              </>
+            )
+            : 'No goods selected'}
+        </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              const isSelected = this.isSelected(good);
+              const setSelected = () => this.setSelected(good);
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              return (
+                <tr
+                  data-cy="Good"
```

Comment:
I couldn't find a key added to the `<tr>...</tr>`

Reply:
My bad